packages/flutter_chat/lib/src/flutter_chat_entry_widget.dart

```diff
@@ -61,24 +57,27 @@ class _FlutterChatEntryWidgetState extends State<FlutterChatEntryWidget> {
   @override
   void initState() {
     super.initState();
-    chatService ??= widget.chatService ?? ChatService();
+    chatService = ChatService(
+      userId: widget.userId,
+      chatRepository: widget.options?.chatRepository,
+      userRepository: widget.options?.userRepository,
+    );
   }
 
   @override
   Widget build(BuildContext context) => InkWell(
         onTap: () async =>
             widget.onTap?.call() ??
-            Navigator.of(context).pushReplacement(
+            Navigator.of(context).push(
               MaterialPageRoute(
                 builder: (context) => FlutterChatNavigatorUserstory(
                   userId: widget.userId,
-                  chatService: chatService,
-                  chatOptions: widget.options,
+                  options: widget.options ?? ChatOptions(),
                 ),
               ),
             ),
         child: StreamBuilder<int>(
-          stream: chatService!.getUnreadMessagesCount(userId: widget.userId),
+          stream: chatService!.getUnreadMessagesCount(),
```

Comment:
If you are always assigning the chat in the initState, then there is no reason for a null check here. You should make it late if it is mutable or late final if you never expect it to change.

Given its dependency on the widget.userId and widget.options, it probably should be mutable and be updated if the parent data would ever change.